src/main/java/org/embulk/input/jira/JiraInputPlugin.java

```diff
@@ -0,0 +1,216 @@
+                int currentPage = 0;
+                int totalCount = jiraClient.getTotalCount(task);
+                int totalPage = JiraUtil.calculateTotalPage(totalCount, MAX_RESULTS);
+                LOGGER.info(String.format("Total pages (%d)", totalPage));
+                while (currentPage < totalPage) {
+                    LOGGER.info(String.format("Fetching page %d/%d", (currentPage + 1), totalPage));
+                    List<Issue> issues = jiraClient.searchIssues(task, (currentPage * MAX_RESULTS), MAX_RESULTS);
+                    for (Issue issue : issues) {
+                        issue.toRecord();
+                        JiraUtil.addRecord(issue, schema, task, pageBuilder);
+                    }
+                    currentPage++;
+                }
+            }
+            pageBuilder.finish();
+        }
+        return Exec.newTaskReport();
+    }
+
+    private JiraClient getJiraClient()
+    {
+        return new JiraClient();
+    }
+
+    @Override
+    public ConfigDiff guess(ConfigSource config)
+    {
+        // Reset columns in case already have or missing on configuration
+        config.set("columns", new ObjectMapper().createArrayNode());
+        ConfigSource guessConfig = createGuessConfig();
+        GuessExecutor guessExecutor = Exec.getInjector().getInstance(GuessExecutor.class);
+        PluginTask task = config.loadConfig(PluginTask.class);
+        JiraClient jiraClient = getJiraClient();
+        List<Issue> issues = jiraClient.searchIssues(task, 0, GUESS_RECORDS_COUNT);
+        issues.stream().forEach(issue -> issue.toRecord());
+        Set<String> uniqAtrribtes = getUniqueAttributes(issues);
+        JsonArray samples = createSamples(issues, uniqAtrribtes);
+        Buffer sample = Buffer.copyOf(samples.toString().getBytes());
+        JsonNode columns = guessExecutor.guessParserConfig(sample, Exec.newConfigSource(), guessConfig).getObjectNode().get("parser").get("columns");
+        ConfigDiff configDiff = Exec.newConfigDiff();
+        configDiff.set("columns", columns);
+        return configDiff;
+    }
+
+    private ConfigSource createGuessConfig()
+    {
+        ConfigSource configSource = Exec.newConfigSource();
+        configSource.set("guess_plugins", new ObjectMapper().createArrayNode().add("jsonpath"));
+        configSource.set("guess_sample_buffer_bytes", GUESS_BUFFER_SIZE);
+        return configSource;
+    }
+
+    private SortedSet<String> getUniqueAttributes(List<Issue> issues)
+    {
+        SortedSet<String> uniqAttributes = new TreeSet<>();
+        issues.stream()
+        .forEach(issue -> {
+            for (Entry<String, JsonElement> entry : issue.getFlatten().entrySet()) {
+                uniqAttributes.add(entry.getKey());
+            }
+        });
+        return uniqAttributes;
+    }
+
+    private JsonArray createSamples(List<Issue> issues, Set<String> uniqAtrribtes)
+    {
+        JsonArray samples = new JsonArray();
+        issues.stream()
```

Comment:
List already has `forEach`, if we want this to be idiomatic stream, then:
```
return issues.stream()
        .map(Issue::getFlatten)
        .map(flattened -> {
            JsonObject unified = new JsonObject();
            for (String attr : uniqAtrribtes ) {
                unified.add(attr, flattened.has(attr) ? flattened.get(attr) : JsonNull.INSTANCE);
            }
            return unified;
        })
        .collect(Collector.of(
                JsonArray::new, JsonArray::add,
                (acc1, acc2) -> { acc1.addAll(acc2); return acc1; }));
```
We generally should avoid mixing streams with side-effects if possible.

Reply:
Changed to for in